scripts/bootstrap_dsh.py

```
@@ -0,0 +1,327 @@
+
+def write_overlay(project_root: Path, *, force_overlay: bool) -> dict[str, Any]:
+    overlay = (project_root / OVERLAY_RELATIVE).resolve()
+    overlay.parent.mkdir(parents=True, exist_ok=True)
+    desired = overlay_text(PLUGIN_ENTRY)
+    backup_path: Path | None = None
+    if overlay.exists():
+        current = overlay.read_text(encoding="utf-8")
+        if current == desired:
+            return {"written": False, "path": json_safe_path(overlay), "status": "unchanged"}
+        if not force_overlay:
+            raise RuntimeError(f"Overlay already exists with different content: {overlay}. Re-run with --force-overlay to replace it.")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)